src/java.base/share/classes/jdk/internal/foreign/LayoutPath.java

```diff
@@ -314,15 +319,27 @@ private static LayoutPath derefPath(MemoryLayout layout, MethodHandle handle, La
 
     // Helper methods
 
-    private void check(Class<?> layoutClass, String msg) {
+    private SequenceLayout requireSequenceLayout() {
+        return requireLayoutType(SequenceLayout.class, "sequence");
+    }
+
+    private GroupLayout requireGroupLayout() {
+        return requireLayoutType(GroupLayout.class, "group");
+    }
+
+    private <T extends MemoryLayout> T requireLayoutType(Class<T> layoutClass, String name) {
         if (!layoutClass.isAssignableFrom(layout.getClass())) {
-            throw badLayoutPath(msg);
+            throw badLayoutPath(
+                    String.format("attempting to select a %s element from a non-%s layout: %s (%s)",
```

Comment:
While it is good to see the breadcrumbs, having them enclosed in parentheses creates confusion as the layouts themselves contain parentheses as shown in the example below.

Is there a better way (e.g. using `{}`)?

```
Bad layout path: attempting to select a sequence element from a non-sequence layout: [i4(3a)i4(3b)](2) ([[2:[i4(3a)i4(3b)](2)](1)](0) -> [2:[i4(3a)i4(3b)](2)](1) -> [i4(3a)i4(3b)](2))
```

Reply:
Maybe instead of parenthesis, you could have something like : `path =` in the message? Or maybe something like
```
Bad layout path: attempting to select a sequence element from a non-sequence layout: [i4(3a)i4(3b)](2), selected from: [2:[i4(3a)i4(3b)](2)](1), selected from: [[2:[i4(3a)i4(3b)](2)](1)](0)
```